src/http/request_context.rs

```diff
@@ -27,20 +29,34 @@ impl Default for RequestContext {
 
 impl RequestContext {
   pub fn new(http_client: Arc<dyn HttpClient>, server: Server, upstream: Upstream) -> Self {
-    Self { req_headers: HeaderMap::new(), http_client, server, upstream, min_max_age: Arc::new(Mutex::new(None)) }
+    Self {
+      req_headers: HeaderMap::new(),
+      http_client,
+      server,
+      upstream,
+      min_max_age: Arc::new(Mutex::new(None)),
+      cache_private: Arc::new(Mutex::new(None)),
+    }
   }
 
   pub async fn execute(&self, req: reqwest::Request) -> anyhow::Result<Response> {
     self.http_client.execute(req).await
   }
-  fn set_min_max_age_conc(&self, min_max_age: u64) {
+  fn set_min_max_age_conc(&self, min_max_age: i64) {
     *self.min_max_age.lock().unwrap() = Some(min_max_age);
   }
-  pub fn get_min_max_age(&self) -> Option<u64> {
+  pub fn get_min_max_age(&self) -> Option<i64> {
```

Comment:
why change it from unsigned? will we ever have negative values here?

Reply:
Yes, @ologbonowiwi. In case of no-cache, it is set to -1.